presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardRecoveryManager.java

```diff
@@ -129,7 +129,11 @@ private void enqueueMissingShards()
     {
         Set<UUID> shardUuids = shardManager.getNodeShards(nodeIdentifier);
         ImmutableSet.Builder<UUID> missingShards = ImmutableSet.builder();
-        shardUuids.stream().filter(shardUuid -> !storageService.getStorageFile(shardUuid).exists()).forEach(missingShards::add);
+        shardUuids.stream().filter(shardUuid -> {
+            File storageFile = storageService.getStorageFile(shardUuid);
+            File backupFile = storageService.getBackupFile(shardUuid);
+            return !(storageFile.exists() && storageFile.length() == backupFile.length());
+        }).forEach(missingShards::add);
```

Comment:
While you're changing this code, update it to use a stream on the original with `Collectors.toSet()`